samples/core/guide/autograph_control_flow.ipynb

```diff
@@ -0,0 +1,770 @@
+        "def nearest_odd_square(x):\n",
+        "  if x > 0:\n",
+        "    x = x * x\n",
+        "    if x % 2 == 0:\n",
+        "      x = x + 1\n",
+        "  return x\n",
+        "\n",
+        "with tf.Graph().as_default():  \n",
+        "  with tf.Session() as sess:\n",
+        "    print(sess.run(nearest_odd_square(tf.constant(4))))\n",
+        "    print(sess.run(nearest_odd_square(tf.constant(5))))\n",
+        "    print(sess.run(nearest_odd_square(tf.constant(6))))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "jXAxjeBr1qWK",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### While loop"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "ucmZyQVL03bF",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "@autograph.convert()\n",
+        "def square_until_stop(x, y):\n",
+        "  while x < y:\n",
+        "    x = x * x\n",
+        "  return x\n",
+        "    \n",
+        "with tf.Graph().as_default():  \n",
+        "  with tf.Session() as sess:\n",
+        "    print(sess.run(square_until_stop(tf.constant(4), tf.constant(100))))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "FXB0Zbwl13PY",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Break from loop"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "1sjaFcL717Ig",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "@autograph.convert()\n",
+        "def argwhere_cumsum(x, threshold):\n",
+        "  current_sum = 0.0\n",
+        "  idx = 0\n",
+        "  for i in range(len(x)):\n",
+        "    idx = i\n",
+        "    if current_sum >= threshold:\n",
+        "      break\n",
+        "    current_sum += x[i]\n",
+        "  return idx\n",
+        "\n",
+        "N = 10\n",
+        "with tf.Graph().as_default():  \n",
+        "  with tf.Session() as sess:\n",
+        "    idx = argwhere_cumsum(tf.ones(N), tf.constant(float(N/2)))\n",
+        "    print(sess.run(idx))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "4LfnJjm0Bm0B",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Advanced example: A training, loop in-graph\n",
+        "\n",
+        "Writing control flow in AutoGraph is easy, so running a training loop in a TensorFlow graph should be easy as well!  \n",
+        "\n",
+        "Here, we show an example of training a simple Keras model on MNIST, where the entire training process -- loading batches, calculating gradients, updating parameters, calculating validation accuracy, and repeating until convergence -- is done in-graph."
```

Comment:
WDYT about a sentence somewhere ahead of this that mentions something in the lines of "oh, and you can use Keras libraries in AutoGraph as well"? Not sure if it's something the user would expect or not.

Reply:
Done. I'll link to the other notebook here when it's ready.